in-an-ocp-cluster/container-image/Dockerfile

```
@@ -19,7 +19,7 @@ FROM ${RHEL_COREOS_8_IMAGE}
 ARG KERNEL_VERSION
 
 COPY --from=builder /etc/driver-toolkit-release.json /etc/
-COPY --from=builder /lib/modules/${KERNEL_VERSION}/ /usr/lib/modules/${KERNEL_VERSION}/
+COPY --from=builder /lib/modules/${KERNEL_VERSION}/*.ko /usr/lib/modules/${KERNEL_VERSION}/
 
 # This is needed in order to autoload the module at boot time.
-RUN echo simple_kmod > /etc/modules-load.d/simple_kmod.conf
+RUN depmod && echo simple_kmod > /etc/modules-load.d/simple_kmod.conf
```

```suggestion
RUN depmod -F /lib/modules/${KERNEL_VERSION}/System.map ${KERNEL_VERSION}
RUN echo simple_kmod > /etc/modules-load.d/simple_kmod.conf
```

I am not sure why, but `depmod`  with its default values isn't working:
```
[2/2] STEP 5/5: RUN depmod && echo simple_kmod > /etc/modules-load.d/simple_kmod.conf
depmod: ERROR: could not open directory /lib/modules/4.18.0-383.el8.x86_64: No such file or directory
depmod: FATAL: could not search modules: No such file or directory
Error: error building at STEP "RUN depmod && echo simple_kmod > /etc/modules-load.d/simple_kmod.conf": error while running runtime: exit status 1
```
The kernel RPMs in the image are `4.18.0-372.26.1.el8_6.x86_64` and I have no idea why `depmod` is using `4.18.0-383.el8.x86_64` by default.

---

I just took the command from here: https://github.com/openshift-psap/simple-kmod/blob/e8528529b810be4e856e741ea2d2f51ea363795a/Makefile#L23

---

Ah yeah it's because `depmod` uses the running kernel by default.  Need to use `depmod -a ${KERNEL_VERSION}`.

---

Sorry for late response. We had a public holiday just after the weekend.

This indeed works.

Would you like to fix that PR and I will merge? (I don't mind doing it myself but then you won't get credit for the change :) )

---

sure, done